From a0c548531df699d0aad7edf2f2c0eff19d10d1de Mon Sep 17 00:00:00 2001
From: Aaradhana Sahu <quic_aarasahu@quicinc.com>
Date: Sat, 4 Nov 2023 13:10:25 +0530
Subject: [PATCH] ath11k: Removing extra flag from MHI

Removed the MHI SBL mode from driver. Since it is not required for host
driver.

Signed-off-by: Gautham Kumar Senthilkumaran <quic_gauthamk@quicinc.com>
---
 drivers/net/wireless/ath/ath11k/mhi.c | 17 -----------------
 1 file changed, 17 deletions(-)

diff --git a/drivers/net/wireless/ath/ath11k/mhi.c b/drivers/net/wireless/ath/ath11k/mhi.c
index b6eda9b..6baeecc 100644
--- a/drivers/net/wireless/ath/ath11k/mhi.c
+++ b/drivers/net/wireless/ath/ath11k/mhi.c
@@ -339,18 +339,6 @@ static void ath11k_mhi_op_write_reg(struct mhi_controller *mhi_cntrl,
 	writel(val, addr);
 }
 
-static void ath11k_mhi_qrtr_instance_set(struct mhi_controller *mhi_cntrl)
-{
-	struct ath11k_base *ab = dev_get_drvdata(mhi_cntrl->cntrl_dev);
-
-	if (ab->hw_rev == ATH11K_HW_QCN9074_HW10) {
-		ath11k_mhi_op_write_reg(mhi_cntrl,
-					mhi_cntrl->bhi + BHI_ERRDBG2,
-					FIELD_PREP(QRTR_INSTANCE_MASK,
-					ab->qmi.service_ins_id - ab->hw_params.qmi_service_ins_id));
-	}
-}
-
 static char *ath11k_mhi_op_callback_to_str(enum mhi_callback reason)
 {
 	switch (reason) {
@@ -372,8 +360,6 @@ static char *ath11k_mhi_op_callback_to_str(enum mhi_callback reason)
 		return "MHI_CB_FATAL_ERROR";
 	case MHI_CB_BW_REQ:
 		return "MHI_CB_BW_REQ";
-	case MHI_CB_EE_SBL_MODE:
-		return "MHI_CB_EE_SBL_MODE";
 	default:
 		return "UNKNOWN";
 	}
@@ -395,9 +381,6 @@ static void ath11k_mhi_op_status_cb(struct mhi_controller *mhi_cntrl,
 		if (!(test_bit(ATH11K_FLAG_UNREGISTERING, &ab->dev_flags)))
 			queue_work(ab->workqueue_aux, &ab->reset_work);
 		break;
-	case MHI_CB_EE_SBL_MODE:
-		ath11k_mhi_qrtr_instance_set(mhi_cntrl);
-		break;
 	default:
 		break;
 	}
-- 
2.17.1

